### ISSUE TITLE
editEdge is not called when I select an edge and click on the 'Edit Edge' button.

### ISSUE THREAD

**Issue by lcahlander** [almende/vis#1756]
I am popping up a modal dialog when I am editing a node.  I want to do the same for editing an edge.

**Comment by josdejong** [comment]
The "Edit Edge" button will just let you connect and edge with different nodes. A callback allowing you to open a dialog to change for example the title is missing, that would be a nice addition.

For now, you could either see if you can hack an extra button into the menu, or use for example a doubleClick event to trigger your dialog.

**Comment by juandjara** [comment]
I am facing this problem too, and what I've done is to put a button that is displayed when the `selectEdge` event is fired on the network and hidden when the `deselectEdge` event is fired.

From there, you get the selected nodeIds and edgeIds.

You can get the edge for an id with `network.body.data.edges.get(edgeId)` and pass it to a function like your `onEditEdge` where you toggle a popup for editing. Along with data, you pass a callback that will update the edge if called with `network.body.data.edges.update(newEdge)`

Is a little weird. I can share a gist if needed. Sorry for my english.

**Comment by AlexDM0** [comment]
Hi, 

EditEdge is called after you drag it. To get around this, a callback should be added for the startEditEdge case to see if you are allowed to edit that specific edge. 

I cannot discourage using the networks internal structure strongly enough. Doing that might break your code without warning in new releases. You can use your own dataset to get the exact same functionality. 

Regards

**Comment by juandjara** [comment]
Thanks for the clarification. So how can I use this startEditEdge method ?? I haven't seen it on the documentation

**Comment by AlexDM0** [comment]
It's not. It's still a concept. Feel like adding it?

Regards

**Comment by juandjara** [comment]
Yeah, I would like to, but I don't know where to start. If I manage to fix this I will send a PR.

Regards.

**Comment by eymiha** [comment]
I worked code in for editEdgeMode to short-circuit the edge source/target changing and go straight to calling an editEdge function, but I can see where you'd want it both ways - that is

  1. Skip the dragging and directly call the editEdge function (my changes), or
  2. Do the dragging and then call the editEdge function (existing functionality).

It appears it may need a boolean to discriminate between the two...
```
manipulation: {
    editEdge: {
        function: userSuppliedEditEdgeFunction,
        adjustEndpoints: false
    }
}
```
to get the new behavior.

Thoughts?

**Comment by mojoaxel** [comment]
fixed with #2329
